atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1900,6 +1900,35 @@ private void putMetadataWithoutChangingSettings(final TableReference tableRef, f
         delete(AtlasDbConstants.DEFAULT_METADATA_TABLE, oldVersions);
     }
 
+
+    @Override
+    public void deleteRange(final TableReference tableRef, final RangeRequest range) {
+        if (range.equals(RangeRequest.all())) {
+            try {
```

Comment:
Ah okay, because you only want to then attempt deleteRange (instead of truncate) on certain styles of truncate failures. If the answer to my other question re: do we want to retry is yes we could just pull that logic into another method, but I guess it doesn't really matter.